### PR TITLE
Added features and fixed stuff

### PR DESCRIPTION
**Pull request by GeorgSteyn**
- **Fixed & Updated: .gitingore**  (Not in root of project and updated file )
- **Updated: Command-Line Arguments** (--init argument no need for =True)
- **Updated: File Handling Path** (Save path for Images, Test Cases and Test Suites)
- **Added Function for Safe Dir Create** (Function to safely work with path)
- **Fixed Issue: Lackey's misleading errors** (Fixed errors and handling of errors given by Lackey)
- **Updated: package.json**
- **Updated: Dashboard Screen** (Does something more useful now!!!)

### CODE REVIEW

File: backend/modules/goodxtest.py [hunk 1]
```diff
@@ -243,9 +252,8 @@ def mouseReleaseEvent(self, event):
             y2 = max(self.begin.y(), self.end.y())
 
             img = ImageGrab.grab(bbox=(x1, y1, x2, y2))
-            print('Wooohooo !!!!')
-            print('ImageName')
```
Comment (Cl3anKillx) [inline backend/modules/goodxtest.py]: Sit terug

File: frontend/src/app/layout/dashboard/dashboard.component.html [hunk 2]
```diff
@@ -1,3 +1,35 @@
 <div class="content" [@routerTransition]>
-    <h1>Dashboard !!!</h1>
+    <div class="jumbotron">
+        <h1 class="display-4">Testing System Dashboard</h1>
+        <hr class="my-4">
+        <h3>
+            <u>Here is all your test counts:</u>
```
Comment (Cl3anKillx) [inline frontend/src/app/layout/dashboard/dashboard.component.html]: Vernader die net na `Test Counts:`

File: frontend/src/app/layout/dashboard/dashboard.component.scss [hunk 3]
```diff
@@ -0,0 +1,10 @@
+.hover-success:hover {
+    transition: 0.25s ease-in;
+    box-shadow: 0 0 20px limegreen !important;
+    opacity: 60%;
+    cursor: pointer;
+}
+
+.hover-success {
+    transition: 1s ease-out;
+}
```
Comment (Cl3anKillx) [inline frontend/src/app/layout/dashboard/dashboard.component.scss]: newline

File: frontend/src/app/layout/dashboard/dashboard.component.html [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+                            <h4 class="card-title font-weight-bold">Test Case</h4>
+                            <p class="card-text">{{ testCaseCount }}</p>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+        <hr class="my-4">
+    </div>
 </div>
```
Comment (Cl3anKillx) [inline frontend/src/app/layout/dashboard/dashboard.component.html]: newline

File: frontend/src/app/layout/dashboard/dashboard.component.ts [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
+    testSuiteCount: string = '0'
+    testCaseCount: string = '0'
+
+    constructor(private backend: BackendService) { }
+
+    ngOnInit() {
+        this.backend.getTestsCount().then(res => this.testCaseCount = res.data);
+        this.backend.getSuitesCount().then(res => this.testSuiteCount = res.data);
+    }
+}
```
Comment (Cl3anKillx) [inline frontend/src/app/layout/dashboard/dashboard.component.ts]: newline

File: frontend/src/app/shared/backend/backend.service.ts [hunk 6]
```diff
@@ -81,6 +81,14 @@ export class BackendService {
         return this.post('getTests');
     }
```
Comment (Cl3anKillx) [inline frontend/src/app/shared/backend/backend.service.ts]: comment, dink ons moet begin comments sit by die funksies. Sal dit maar so sistematies doen.

File: frontend/src/app/shared/backend/backend.service.ts [hunk 7]
```diff
@@ -81,6 +81,14 @@ export class BackendService {
...
 
+    getTestsCount() {
+        return this.post('getTestsCount');
+    }
+
```
Comment (Cl3anKillx) [inline frontend/src/app/shared/backend/backend.service.ts]: comment, dink ons moet begin comments sit by die funksies. Sal dit maar so sistematies doen.

File: backend/modules/goodxtest.py (hunk 1 above)
Comment (GeorgSteyn) [reply backend/modules/goodxtest.py]: Ek het iets rerig special vir jou gelos daarso!!!

**Pull request merged**